src/index.tsx

```diff
@@ -4,12 +4,16 @@ import './index.css';
 import './styles/bootstrap_replacement.css';
 import App from './App';
 import {Auth} from './libs/auth/auth';
+import {OidcBroker} from './libs/auth/oidcBroker';
 import {unregister} from './registerServiceWorker';
 import {BrowserRouter} from 'react-router-dom';
 
 const load = async () => {
   unregister();
   await Auth.initialize();
+  if (window.location.pathname.startsWith('/redirect-from-oauth')) {
+    await OidcBroker.getUserManager().signinPopupCallback(window.location.href);
```

Comment:
what does this do?

Reply:
Great question. In the original version of this work, there was a [whole component](https://github.com/DataBiosphere/duos-ui/pull/2611/files#diff-326a2368c43259048d25d6a6260488fed18c66c82678ef208bd6aa99020ed902) dedicated to servicing this single path. This is my attempt to simplify it. In short, it does this:

1. The OIDC client calls a popup window with a redirect url here: https://github.com/DataBiosphere/duos-ui/pull/2667/files#diff-c22f0b724cae791bb887d11ce9ada280a1581625e133634e5c2356a5637d6980L42
2. The user goes through the auth dance and is redirected to that url, but the popup doesn't go away.
3. On reload, we detect the path
4. Call the `signinPopupCallback` which does two things
    - It closes the open popup
    - Redirects the user to the original href, which is either the home page, or a sub-page inside the app to which we redirect.

Reply:
I'm going to add some comments around this for future devs.